build_runner/lib/src/package_graph/apply_builders.dart

```diff
@@ -28,29 +28,58 @@ PackageFilter toPackages(Set<String> packages) =>
 PackageFilter toAll(Iterable<PackageFilter> filters) =>
     (p) => filters.any((f) => f(p));
 
-/// Indcates that [builder] should be run against all packages matching
+typedef Builder BuilderFactory(List<String> args);
+
+/// Apply [builder] to the root package.
+BuilderApplication applyToRoot(Builder builder,
+        {List<String> inputs, List<String> excludes}) =>
+    new BuilderApplication._('', '', [(_) => builder], null,
+        inputs: inputs, excludes: excludes);
+
+/// Apply each builder from [builderFactories] to the packages matching
 /// [filter].
 ///
 /// If the builder should only run on a subset of files within a package pass
 /// globs to [inputs] or [excludes];
```

Comment:
I think we lost any public docs on the `isOptional` arg during this transition, it would be good to add something back since the behavior is a bit nuanced.

Reply:
Done